xbmc/filesystem/CurlFile.cpp

```diff
@@ -645,6 +646,14 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
   // Set the lowspeed time very low as it seems Curl takes much longer to detect a lowspeed condition
   g_curlInterface.easy_setopt(h, CURLOPT_LOW_SPEED_TIME, m_lowspeedtime);
 
+  // enable tcp keepalive
+  if (m_keepAliveInterval == 30)
```

Comment:
I'd just make this
`
if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_curlKeepAliveInterval != 0)
 {
  g_curlInterface.easy_setopt(h, CURLOPT_TCP_KEEPALIVE, 1);
  g_curlInterface.easy_setopt(h, CURLOPT_TCP_KEEPIDLE, m_keepAliveInterval / 2);
  g_curlInterface.easy_setopt(h, CURLOPT_TCP_KEEPINTVL, m_keepAliveInterval); 
}

Reply:
yep, that's what I had in mind as well